SerialPrograms/Source/PokemonSV/Programs/Farming/PokemonSV_BlueberryQuests.cpp

```diff
@@ -0,0 +1,1205 @@
+
+    return OCR::read_number(console, BP_value);
+}
+
+std::vector<BBQuests> read_quests(const ProgramInfo& info, ConsoleHandle& console, BotBaseContext& context, const BBQOption& BBQ_OPTIONS) {
+    std::vector<BBQuests> quest_list;
+
+    //Open quest list. Wait for it to open.
+    WhiteButtonWatcher right_panel(COLOR_BLUE, WhiteButton::ButtonB, {0.484, 0.117, 0.022, 0.037});
+    int result = run_until(
+        console, context,
+        [&](BotBaseContext& context){
+            for (int i = 0; i < 6; i++) { //try 6 times
+                pbf_press_dpad(context, DPAD_RIGHT, 50, 20);
+                pbf_wait(context, 200);
+                context.wait_for_all_requests();
+            }
+        },
+        {{ right_panel }}
+    );
+    if (result == 0){
+        console.log("Found quest panel.");
+    }
+    context.wait_for_all_requests();
+
+    //Read in the initial 4 quests.
+    console.log("Reading quests.");
+    VideoSnapshot screen = console.video().snapshot();
+    BlueberryQuestDetector first_quest_detector(console.logger(), COLOR_GREEN, BBQ_OPTIONS.LANGUAGE, BlueberryQuestDetector::QuestPosition::FIRST);
+    BlueberryQuestDetector second_quest_detector(console.logger(), COLOR_GREEN, BBQ_OPTIONS.LANGUAGE, BlueberryQuestDetector::QuestPosition::SECOND);
+    BlueberryQuestDetector third_quest_detector(console.logger(), COLOR_GREEN, BBQ_OPTIONS.LANGUAGE, BlueberryQuestDetector::QuestPosition::THIRD);
+    BlueberryQuestDetector fourth_quest_detector(console.logger(), COLOR_GREEN, BBQ_OPTIONS.LANGUAGE, BlueberryQuestDetector::QuestPosition::FOURTH);
+
+    quest_list.push_back(BBQuests_string_to_enum(first_quest_detector.detect_quest(screen)));
+    quest_list.push_back(BBQuests_string_to_enum(second_quest_detector.detect_quest(screen)));
+    quest_list.push_back(BBQuests_string_to_enum(third_quest_detector.detect_quest(screen)));
+    
+    std::string fourth_quest = fourth_quest_detector.detect_quest(screen);
+    if (fourth_quest != "") {
+        quest_list.push_back(BBQuests_string_to_enum(fourth_quest));
+    }
+
+    //Close quest list
+    pbf_mash_button(context, BUTTON_B, 100);
+    context.wait_for_all_requests();
+
+    return quest_list;
+}
+
+std::vector<BBQuests> process_quest_list(const ProgramInfo& info, ConsoleHandle& console, BotBaseContext& context, const BBQOption& BBQ_OPTIONS, std::vector<BBQuests>& quest_list, uint8_t& eggs_hatched) {
+    std::vector<BBQuests> quests_to_do;
+
+    console.log("Processing quests.");
+    //Put all do-able quests into a different list
+    for (auto n : quest_list) {
+        if (not_possible_quests.contains(n)) {
+            console.log("Quest not possible");
+        }
+        else {
+            //Check eggs remaining.
+            if (n == BBQuests::hatch_egg && eggs_hatched >= BBQ_OPTIONS.NUM_EGGS) {
+                console.log("Out of eggs! Quest not possible.");
+                if (BBQ_OPTIONS.OUT_OF_EGGS == BBQOption::OOEggs::Reroll) {
```

Comment:
If you make this a switch statement so that `BBQ_OPTIONS.OUT_OF_EGGS` is only read once, you can unlock this config.